sklearn/preprocessing/tests/test_data.py

```diff
@@ -1533,6 +1533,27 @@ def test_quantile_transform_nan():
     assert not np.isnan(transformer.quantiles_[:, 1:]).any()
 
 
+@pytest.mark.parametrize("sparse_data", [False, True])
+def test_quantile_transformer_sorted_quantiles(sparse_data):
+    # Non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/15733
+    # Taken from upstream bug report:
+    # https://github.com/numpy/numpy/issues/14685
+    X = np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9, 8, 8, 7] * 10)
```

Comment:
Shoot, I try to produce the failure running differnent size and way of generating unsuccessfully.

Reply:
Nice that you got one

Reply:
The trick was to make that dataset larger than 100 samples, otherwise `quantiles_` was limited by `X.shape[0]`.

So I just duplicated the samples 10 times and the monotonicity issue as still present :)

Reply:
BTW, I am not sure if `quantile_transformer.quantiles_.shape[0]` being smaller than `quantile_transformer.n_quantiles` (when the training set size is too small) is a bug or not.

Reply:
We raise warning in this case so I think that this is fine (at least we expected it)